tc/core/polyhedral/scop.cc

```diff
@@ -440,6 +440,89 @@ detail::ScheduleTree* obtainOuterBand(detail::ScheduleTree* root) {
   }
   return tree;
 }
+
+// Check if there is at least one band with a coincident dimension
+// in the tree.
+bool hasAtLeastOneCoincidentLoop(detail::ScheduleTree* tree) {
+  if (auto band = tree->elemAs<ScheduleTreeElemBand>()) {
+    if (find(band->coincident_.begin(), band->coincident_.end(), true) !=
```

Comment:
This looks more complex than necessary, just hoist the `permutable_` condition and make a simple for loop?

```
if (auto band = tree->elemAs<ScheduleTreeElemBand>() && band->permutable_) {
   for (auto c : band->coincident_) {
      if (c) {
         return true;
      }
   }
```